src/wallet/wallet.cpp

```diff
@@ -5586,24 +5586,63 @@ bool CWallet::InitLoadWallet() {
         nWalletDBUpdated++;
 
         // Restore wallet transaction metadata after -zapwallettxes=1
-        if (GetBoolArg("-zapwallettxes", false) && GetArg("-zapwallettxes", "1") != "2") {
-            CWalletDB walletdb(walletFile);
+        if (GetBoolArg("-zapwallettxes", false)) {
```

Comment:
We probably need to do something similar for sigma. Could you please create a PR for sigma branch, or just add a trello card?

Reply:
Here it is: https://trello.com/c/KNSHlfOd/487-zapwallettxes-should-correctly-restore-status-of-sigma-spends